src/NBitcoin/Consensus.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using NBitcoin.BouncyCastle.Math;
+
+namespace NBitcoin
+{
+    public enum BuriedDeployments : int
+    {
+        /// <summary>
+        /// Height in coinbase
+        /// </summary>
+        BIP34,
+
+        /// <summary>
+        /// Height in OP_CLTV
+        /// </summary>
+        BIP65,
+
+        /// <summary>
+        /// Strict DER signature
+        /// </summary>
+        BIP66
+    }
+
+    public class Consensus
+    {
+        /// <summary>
+        /// An extension to <see cref="Consensus"/> to enable additional options to the consensus data.
+        /// </summary>
+        public class ConsensusOptions
+        {
+        }
+
+        public ConsensusOptions Options { get; set; }
+
+        public class BuriedDeploymentsArray
+        {
+            readonly int[] heights;
+
+            public BuriedDeploymentsArray()
+            {
+                this.heights = new int[Enum.GetValues(typeof(BuriedDeployments)).Length];
+            }
+
+            public int this[BuriedDeployments index]
+            {
+                get { return this.heights[(int)index]; }
+                set { this.heights[(int)index] = value; }
+            }
+        }
+
+        public class BIP9DeploymentsArray
+        {
+            readonly BIP9DeploymentsParameters[] parameters;
+
+            public BIP9DeploymentsArray()
+            {
+                this.parameters = new BIP9DeploymentsParameters[Enum.GetValues(typeof(BIP9Deployments)).Length];
+            }
+
+            public BIP9DeploymentsParameters this[BIP9Deployments index]
+            {
+                get { return this.parameters[(int)index]; }
+                set { this.parameters[(int)index] = value; }
+            }
+        }
+
+        public Consensus()
+        {
+            this.BuriedDeployments = new BuriedDeploymentsArray();
+            this.BIP9Deployments = new BIP9DeploymentsArray();
+
+            this.ConsensusFactory = new ConsensusFactory()
+            {
+                Consensus = this
+            };
+        }
+
+        public BuriedDeploymentsArray BuriedDeployments { get; set; }
```

Comment:
Please add comments for this and the following fields

Reply:
I totally agree comments will be nice on each of the elements, however we might have time constraints on this PR (for sidechains).
So am happy to create an issue and do it as a separate PR.